### DISCUSSION TITLE
Extract result of travel time inversion to a numpy array

### DISCUSSION THREAD

**Discussion by sminize** [gimli-org/pyGIMLi#372] [Q&A]
Hello everyone,
i want to extract the result of a travel time inversion into a numpy array. The inversion was done in 3D
with the following command.

```
invmodel = tt.invert(data, mesh=mesh, secNodes=3, lam=2500, zWeight=1,
                     useGradient=False, verbose=False,blockyModel=True,robustdata=False)
```

How can i get the inversion result (velocity) in invmodel in a numpy array?

All the best and thank you for your help,

Simon

**Comment by halbmy** [comment]
The results of the invert call is a pyGIMLi vector that is widely compatible with a numpy array. However, it can be transformed into a real array using `invmodel.array()` or `np.array(invmodel)`.

**Comment by sminize** [comment]
Thanks a lot - Afterwards I just need to reshape the data.

Just one more question - Is there also a way to extract the ray paths into a numpy array (in my case in 3D).
I can draw them with : `tt.drawRayPaths(ax=ax2, color="0.8", alpha=0.3)`, but this doesn't work in 3D.

All the best and thank you for your help,

Simon

**Comment by halbmy** [comment]
Please have a look at https://www.pygimli.org/_examples_auto/2_seismics/plot_05_refraction_3D.html where 3D rays are plotted and have a look into the source code https://www.pygimli.org/_modules/pygimli/physics/traveltime/TravelTimeManager.html#TravelTimeManager.drawRayPaths on how the paths can be accessed.
